sdk/python/feast/infra/utils/snowflake_utils.py

```diff
@@ -0,0 +1,276 @@
+import configparser
+import os
+import random
+import string
+from logging import getLogger
+from tempfile import TemporaryDirectory
+from typing import Dict, Iterator, List, Optional, Tuple, cast
+
+import pandas as pd
+import snowflake.connector
+from snowflake.connector import ProgrammingError, SnowflakeConnection
+from snowflake.connector.cursor import SnowflakeCursor
+from tenacity import (
+    retry,
+    retry_if_exception_type,
+    stop_after_attempt,
+    wait_exponential,
+)
+
+from feast.errors import SnowflakeIncompleteConfig, SnowflakeQueryUnknownError
+
+getLogger("snowflake.connector.cursor").disabled = True
+getLogger("snowflake.connector.connection").disabled = True
+getLogger("snowflake.connector.network").disabled = True
+logger = getLogger(__name__)
+
+
+def execute_snowflake_statement(conn: SnowflakeConnection, query) -> SnowflakeCursor:
+    cursor = conn.cursor().execute(query)
+    if cursor is None:
+        raise SnowflakeQueryUnknownError(query)
+    return cursor
+
+
+def get_snowflake_conn(config, autocommit=True) -> SnowflakeConnection:
+    if config.type == "snowflake.offline":
+        config_header = "connections.feast_offline_store"
+
+    config = dict(config)
+
+    # read config file
+    config_reader = configparser.ConfigParser()
+    config_reader.read([config["config_path"]])
+    if config_reader.has_section(config_header):
+        kwargs = dict(config_reader[config_header])
+    else:
+        kwargs = {}
+
+    kwargs.update((k, v) for k, v in config.items() if v is not None)
+
+    try:
+        conn = snowflake.connector.connect(
+            account=kwargs["account"],
+            user=kwargs["user"],
+            password=kwargs["password"],
+            role=f'''"{kwargs['role']}"''',
+            warehouse=f'''"{kwargs['warehouse']}"''',
+            database=f'''"{kwargs['database']}"''',
+            schema=f'''"{kwargs['schema_']}"''',
+            application="feast",
+            autocommit=autocommit,
+        )
+
+        return conn
+    except KeyError as e:
+        raise SnowflakeIncompleteConfig(e)
+
+
+def write_pandas(
```

Comment:
for code that you "copy / pasted + modified", can you add TODOs in the code and assign to yourself to merge?